docs/content/languages/java.md

```diff
@@ -0,0 +1,226 @@
+
+## Before you start
+
+We'll assume you have Fission and Kubernetes setup.  If not, head over
+to the [install guide](../installation/_index.en.md).  Verify your Fission setup with:
+
+```
+$ fission --version
+```
+
+## Add JVM environment to your cluster
+
+Fission language support is enabled by creating an _Environment_.  An environment is the language-specific part of Fission.  It has a container image in which your function will run.
+
+```
+$ fission environment create --name java --image fission/jvm-env --builder fission/jvm-builder
+```
+
+## Write a simple function in Java
+
+A function needs to implement the `io.fission.Function` class and override the `call` method. The call method receives the `RequestEntity` and `Context` as inputs and needs to return `ResponseEntity` object. Both `RequestEntity` and `ResponseEntity` are from `org.springframework.http` package and provide a fairly high level and rich API to interact with request and response objects. The function code responds with "Hello World" in response body looks as shown below:
+
+```java
+package io.fission;
+
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+
+import io.fission.Function;
+import io.fission.Context;
+
+public class HelloWorld implements Function {
+
+	@Override
+	public ResponseEntity<?> call(RequestEntity req, Context context) {
+		return ResponseEntity.ok("Hello World!");
+	}
+
+}
+```
+
+## HTTP requests and HTTP responses
+
+Java function provides easy access to the Request and Response using Spring framework's `RequestEntity` and `ResponseEntity` objects.
+
+### Accessing HTTP Requests
+
+#### Headers
+
+You can access headers object from the request object and then use various methods on header object to retrieve a specific header or get a collection of all headers.
+
+```java
+	HttpHeaders headers = req.getHeaders();
+	List<String> values = headers.get("keyname");
+```
+
+#### Query string
+
+You can use the URI object in request object and parse the query parameters as shown below.
+
+```java
+	Map<String, String> query_pairs = new LinkedHashMap<String, String>();
+	URI url = req.getUrl();
+	String query = url.getQuery();
+	String[] pairs = query.split("&");
+	for (String pair : pairs) {
+		int idx = pair.indexOf("=");
+		query_pairs.put(URLDecoder.decode(pair.substring(0, idx), "UTF-8"), URLDecoder.decode(pair.substring(idx + 1), "UTF-8"));
+	}
+```
+
+#### Body 
+
+The body of the request object can be accessed as a map. You can use the map to convert to a value object using Jackson library's `ObjectMappper`. You will need to add the Jackson library to dependencies and import appropriate classes.
+
+```java
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+final ObjectMapper mapper = new ObjectMapper();
+HashMap data = (HashMap) req.getBody();
+Data iotData = mapper.convertValue(data, Data.class);
```

Comment:
can you add an example of class Data & curl command here? so that users can understand it more quickly.

Reply:
The data object will depend and vary for each user here - and I don't think we should introduce a concrete object, I will add a comment here.